### PR TITLE
docker: Do not catchup if already initialized.

### PR DESCRIPTION
**Pull request by winder**
## Summary

Avoid running fast catchup in docker containers which have already been initialized.

## Test Plan

1. Build docker container with these changes:
```
docker build \
           -t wwinder/algod:fast-catchup-fix  \
           --build-arg TARGETARCH=amd64 --no-cache \
           --build-arg BRANCH=will/docker-catchup-init \
           --build-arg URL=https://github.com/winder/go-algorand \
           .
```

2. Run a container such that it initialize a local data directory
```
docker run --rm -it -p 8080:8080 --name algod-test-run \
           -e ADMIN_TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa \
           -e TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa \
           -e PROFILE=development \
           -e NETWORK=betanet \
           -e FAST_CATCHUP=1 \
           -v (pwd)/docker_fast:/algod/data \
           wwinder/algod:fast-catchup-fix
```

3. Observe that fast catchup runs, and completes:
```
curl "localhost:8080/v2/status?pretty" -H "Authorization: bearer aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
```

4. Shutdown the container, and start it back up again with the same command from step 2.
5. Monitor status with command from step 3, it should not run fast-catchup.
6. Shutdown the container, start it back up again with `algorand/algod:3.18.0-stable`.
7. Monitor status with command from step 3, it should start fast-catchup (the bug).

### CODE REVIEW

File: cmd/goal/node.go [hunk 1]
```diff
@@ -61,7 +61,7 @@ var newNodeFullConfig bool
 var watchMillisecond uint64
 var abortCatchup bool
 var fastCatchupForce bool
-var initializeCatchup uint64
+var minRounds uint64
```
Comment (bbroder-uji) [inline cmd/goal/node.go:64]:
```suggestion
var minCatchupRounds uint64
```

Comment (winder) [reply cmd/goal/node.go]: Thanks Bob, I renamed the variable to be clearer.

All the globals in these old goal commands really bug me. I've done away with them in the other commands I work on: https://github.com/algorand/go-algorand/blob/master/tools/block-generator/runner/runner.go#L30

Comment (bbroder-uji) [reply cmd/goal/node.go]: Yeah that is a better way to go but meh.

**Comment by bbroder-uji** [comment]
LGTM

File: cmd/goal/node.go [hunk 2]
```diff
@@ -61,7 +61,7 @@ var newNodeFullConfig bool
 var watchMillisecond uint64
 var abortCatchup bool
 var fastCatchupForce bool
-var initializeCatchup uint64
+var minCatchupRounds uint64
```
Comment (algorandskiy) [inline cmd/goal/node.go:64]: thanks for updating. Maybe update REST as well?

**Review by algorandskiy — DISMISSED** [review]

**Review by jannotti — COMMENTED** [review]
How do we manage the releases so this doesn't get run against old algods?

**Comment by winder** [comment]
@jannotti the docker image and algod are released and packaged together, so it isn't an issue.

**Review by algorandskiy — APPROVED** [review]

**Review by jannotti — APPROVED** [review]

**Review by algorandskiy — APPROVED** [review]

File: docker/files/run/run.sh [hunk 3]
```diff
@@ -27,15 +27,8 @@ fi
...
-  local FAST_CATCHUP_URL="https://algorand-catchpoints.s3.us-east-2.amazonaws.com/channel/CHANNEL/latest.catchpoint"
-  local CATCHPOINT=$(curl -s ${FAST_CATCHUP_URL/CHANNEL/$NETWORK})
-  if [[ "$(echo $CATCHPOINT | wc -l | tr -d ' ')" != "1" ]]; then
-    echo "Problem starting fast catchup."
-    exit 1
-  fi
-
   sleep 5
-  goal node catchup "$CATCHPOINT"
+  goal node catchup --force --min 1000000
```
[the remark this answers is not in the document]
Comment (winder) [reply docker/files/run/run.sh:31]: I think this is waiting for algod's API server to start, so still probably necessary.

[the remark this answers is not in the document]
Comment (winder) [reply docker/files/run/run.sh:31]: All the APIs have had the name changed now.

**Pull request merged**